packages/medusa/src/api/admin/direct-upload/route.ts

```diff
@@ -0,0 +1,35 @@
+import { ulid } from "ulid"
+import { MIMEType } from "util"
+import type {
+  MedusaResponse,
+  AuthenticatedMedusaRequest,
+} from "@medusajs/framework/http"
+import type { HttpTypes } from "@medusajs/framework/types"
+import type { AdminDirectUploadRequestPayloadType } from "./validators"
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminDirectUploadRequestPayloadType>,
+  res: MedusaResponse<HttpTypes.AdminDirectUploadSignatureResponse>
+) => {
+  const fileProvider = req.scope.resolve("file")
+  const { filename: originalname, mimeType, size } = req.body
+
+  const type = new MIMEType(mimeType)
+  const extension = type.subtype
+  const uniqueFilename = `${ulid()}.${extension}`
```

Comment:
I think we should allow passing the filename from the UI (and default to ulid) so they can upload files in nested folders (eg. `/media/videos/<ulid>`)

Reply:
Is there any specific reason that frontend would want to do that? Because, we do not display a gallery of files anywhere, so the hierarchy might not make much difference anyways.

Reply:
It's not needed for our current functionalities, it was more if someone wants to reuse the endpoint for some custom content, and they don't want everything flat in the bucket. Eg. in Cloud we structure dumps, some image content, etc. in different folders.

It was just a suggestion though, we can always add it later on if needed